lib/pure/oids.nim

```diff
@@ -74,7 +74,7 @@ proc `$`*(oid: Oid): string =
 
 
 let
-  t = getTime().toUnix.int32
+  t = cast[int32](getTime().toUnix)
```

Comment:
So, I'm a bit confused here - won't this *ignore* overflow, rather than prevent it?

Reply:
The internal uses 32 bits to store time stamp, we have to stick with it.

Only the last 32 bits are considered, which means if the number exceed 4294967296, it is like starting a new round of counting. For `genOid`, it doesn't require monotonically increasing time, an almost unique time stamp is good enough.

Reply:
I feel like this is a limitation that should be corrected - wouldn't it be better to simply modify the OID structure to use a 64-bit integer?

Reply:
As said in the original issue, making the timestamp different than 32 bit would make it incompatible with MongoDB's specification of the structure, which is what this module is based on.

To make the cast "safer" we can do `cast[int32](getTime().toUnix and 0xFFFFFFFF)`.

Reply:
What does MongoDB do in this situation, do you know?

Reply:
At this point, it's fine to abandon the idea of MongoDB interop. I only tried to say that's where the code came from so don't blame me for its misdesigns.

Reply:
I mean this only becomes an issue in 2038.

We can use `cast[int32] ` for the existing Oid structure, then we can add a new Oid64 structure, which is the same but with an int64 timestamp instead of int32.